### DISCUSSION TITLE
C Programlama Dilinde Kullanıcının İsteği Kadar Değişken Oluşturma

### DISCUSSION THREAD

**Discussion by alpeerkaraca** [19/classroom#56] [C] [answered]
Sorum başlıkta olduğu gibi. Yaptığım bir programda kullanıcının girdiği veriler liste şeklinde hafıza edilecek ancak kullanıcı verileri farklı listeler altında tutmak isterse istediği miktarda listeyi oluşturmasını istiyorum. İnternette araştırdığım zaman bu konuda pek bir bilgi bulamadım. Mümkünse gerekli dokümantasyon ya da yolu anlatır mısınız? Şimdiden teşekkür ederim.

**Comment by roktas** [comment, marked as answer]
Anladığım kadarıyla cevaplamaya çalışayım (sorun çok net değil).  Değişkenler (C gibi bir dilde) statik olarak, yani program çalışmadan önce derleme zamanında oluşturulan ve yer ayrılan "şey"lerdir; çalışma zamanında (runtime) "değişken oluşturmak" gibi bir şey olamaz. 
 Sana gereken "istenildiği kadar değişken" değil istenilen sayıda "liste" tutacak bir "liste".  Değişken yerine bir dizinin/listenin ögeleri olarak düşün konuyu.

Peki "bir listeler listesi"ni dinamik olarak nasıl oluşturacaksın?  Her şey değişkenlik gösteriyorsa, yani gerek liste sayısı, gerekse de her bir listedeki veri sayısı önceden bilinmiyorsa bir "bağlı liste" ("linked list" olarak arat) kullanabilirsin, öyle ki linked list'teki her bir eleman da bir "linked list" olacak.

**Reply by alpeerkaraca** [reply]
Tamamdır hocam dediğinizi anladım bunun üzerine biraz daha düşüneceğim. Teşekkürler.
